### DISCUSSION TITLE
Relaxing R's naming conventions in `mlr3`

### DISCUSSION THREAD

**Discussion by Fred-Wu** [mlr-org/mlr3#776] [General]
I am trying an old project using mlr3 for learning purposes. The data has hundreds of variables directly downloaded from a database, so the names of those variables may not necessarily follow R's naming conventions. It is also inpractical to rename them.  With many of those variables with non-R names, here is the error message I got. 

```
Error in .__Task__initialize(self = self, private = private, super = super,  : 
  Assertion on 'feature names' failed: Must have names according to R's variable naming conventions, but element 29 does not comply.
```

Also this error will definitely occur if I tries to re-encode data with sparse matrix, like
```
dmy <- caret::dummyVars(" ~ .", data = df)
df <- Matrix(predict(dmy, newdata = df), sparse = T)
```
This will create more unconventional variable names. 

Could `mlr3` relax the naming conventions?

**Comment by be-marc** [comment]
Are you using non-ascii characters?

```r
library(mlr3)
library(data.table)

options(mlr3.allow_utf8_names = TRUE)

data = data.table(target = 1, Æ = 1)
as_task_regr(data, target = "target")
```

Note,

> `"mlr3.allow_utf8_names"`: If set to `TRUE`, checks on the feature names are relaxed, allowing
 non-ascii characters in column names. This is an experimental and temporal option to
pave the way for text analysis, and will likely be removed in a future version of the package.

**Reply by Fred-Wu** [reply]
Some of the longer names have a space or a hyphen in between. 
But I think I can remove those from the names. 

But it is good to know there exists an option to relax the naming conventions.
